### ISSUE TITLE
Update P2.js implementation?

### ISSUE THREAD

**Issue by Cristy94** [photonstorm/phaser#2743]
I've seen that the [p2.js repo](https://github.com/schteppe/p2.js) has been updated and multiple optimizations were included.

Can the new version be included in Phaser? Does phaser have any custom functions created, or can I simply replace the source code with the one from the p2 repo?

**Comment by photonstorm** [comment]
Yeah it should be fine - try it and see! It's on my list to update soon anyway. If there are new features then they won't have Phaser mappings of course, but as long as the underlying API hasn't changed it should 'just work'.

**Comment by schteppe** [comment]
Haven't released for a while, no. The new p2.js stuff needs more testing before a release, I just haven't gotten to it.

@Cristy94 The p2.js master branch has a few minor API changes but should work with Phaser out of the box. If it doesn't, let me know! I'll compile a change log when releasing.

**Comment by Cristy94** [comment]
@schteppe Yeah, it works fine, I built it myself and everything worked as expected. I was a bit confused why the release version was so far behind, I thought there were still unimplemented things that were changed that stopped you from building a new release.

**Comment by photonstorm** [comment]
I'll wait for a proper release I think, so I can go through a change log and see what needs modifying in Phaser.

It's easy enough to replace locally if devs need to upgrade sooner.
